functions/go/set-enforcement-action/main.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	policyAPIVersion     = "constraints.gatekeeper.sh/v1beta1"
+	enforcementActionKey = "enforcementAction"
+	denyActionValue      = "deny"
+	dryRunActionValue    = "dryrun"
+)
+
+//nolint
+func main() {
+	fp := SetEnforcementActionProcessor{}
+	cmd := command.Build(&fp, command.StandaloneEnabled, false)
+	if err := cmd.Execute(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+func (fp *SetEnforcementActionProcessor) Process(resourceList *framework.ResourceList) error {
+	resourceList.Result = &framework.Result{
+		Name: "set-enforcement-action",
+	}
+
+	// get the enforcementAction value from functionConfig
+	var acn string
+	err := getEnforcementAction(resourceList.FunctionConfig, &acn)
+	if err != nil {
+		resourceList.Result.Items = getErrorItem(err.Error())
+		return err
+	}
+
+	// process policies in the package and display results
+	items, err := processPolicies(resourceList.Items, acn)
+	if err != nil {
+		resourceList.Result.Items = getErrorItem(err.Error())
+		return err
+	}
+	resourceList.Result.Items = items
+	return nil
+}
+
+func processPolicies(resourceList []*yaml.RNode, acn string) ([]framework.ResultItem, error) {
+	var resultItems []framework.ResultItem
+	for _, node := range resourceList {
+		if node.IsNilOrEmpty() {
+			continue
+		}
+
+		//check for resources of type policy
+		metadata, err := node.GetMeta()
+		if err != nil {
+			return nil, err
+		}
+
+		if metadata.Name == "" || metadata.Kind == "" || metadata.APIVersion != policyAPIVersion {
```

Comment:
Technically it may not be but all CIS policies have names set and I thought it's better for a policy to have a name instead of skipping it. It also helps us monitor the usage internally when we run our metrics.